openhtf/__init__.py

```diff
@@ -72,37 +90,50 @@ def __call__(self, phase_func):
 PhasePlug = mutablerecords.Record('PhasePlug', ['name', 'cls', 'update_kwargs'])
 
 
-class TestPhaseInfo(mutablerecords.Record(
-    'TestPhaseInfo', ['func', 'source'],
-    {'options': TestPhaseOptions, 'plugs': list, 'measurements': list})):
-  """TestPhase function and related information.
+class PhaseInfo(mutablerecords.Record(
+    'PhaseInfo', ['func', 'source'],
+    {'options': PhaseOptions, 'plugs': list, 'measurements': list})):
+  """Test phase function and framework metadata.
+
+  This is a wrapper around a test phase function that also tracks options,
+  plugs, and measurements to be used on invokation of the phase.  When
+  invoked, plugs are automatically taken out of the phase_data passed in
+  and passed into the phase function as keyword arguments.
 
   Attributes:
     func: Function to be called (with phase_data as first argument).
     source: Source code of func.
-    options: TestPhaseOptions instance.
+    options: PhaseOptions instance.
     plugs: List of PhasePlug instances.
     measurements: List of Measurement objects.
   """
 
   @classmethod
   def WrapOrCopy(cls, func):
-    """Return a new TestPhaseInfo from the given function or instance.
+    """Return a new PhaseInfo from the given function or instance.
 
     We want to return a new copy so that you can reuse a phase with different
     options, plugs, measurements, etc.
 
     Args:
-      func: A phase function or TestPhaseInfo instance.
+      func: A phase function or PhaseInfo instance.
 
     Returns:
-      A new TestPhaseInfo object.
+      A new PhaseInfo object.
     """
     if not isinstance(func, cls):
       return cls(func, inspect.getsource(func))
     # We want to copy so that a phase can be reused with different options, etc.
     return mutablerecords.CopyRecord(func)
 
+  @property
+  def name(self):
+    return self.func.__name__
```

Comment:
How would you specify it for a lambda?  I feel like fixing your callable to be better named via __name__ is probably the recommended fix for that?